packages/panzer/dof-mgr/src/Panzer_GeometricAggFieldPattern.cpp

```diff
@@ -76,54 +77,97 @@ void GeometricAggFieldPattern::buildPattern(const std::vector<Teuchos::RCP<const
              "have the same geometry!");
 
    // copy cell topology
-   cellTopo_ = patterns[0]->getCellTopology();
+   cellTopo_ = patterns[0].second->getCellTopology();
 
    // grab the dimension
-   dimension_ = patterns[0]->getDimension();
+   dimension_ = patterns[0].second->getDimension();
    patternData_.resize(dimension_+1);
 
    // build space for subcells
    std::vector<int> subcellCount(dimension_+1);
    for(std::size_t d=0;d<dimension_+1;d++) {
-      subcellCount[d] = patterns[0]->getSubcellCount(d);
+      subcellCount[d] = patterns[0].second->getSubcellCount(d);
       patternData_[d].resize(subcellCount[d]);
    }
 
-   // build geometric pattern: increment it logically
-   // over all the subcells.
+   // Build geometric pattern: increment it logically over all the
+   // subcells. Start with CG fields first. Then all DG fields. Then
+   // all SKELETON fields. This is done so that we can use offsets
+   // when mapping DOFs to subcells.
    int counter = 0;
    for(std::size_t d=0;d<dimension_+1;d++) {
       for(int s=0;s<subcellCount[d];s++) {
          std::vector<int> & current = patternData_[d][s];
          for(std::size_t p=0;p<patterns.size();p++) {
-            RCP<const FieldPattern> field = patterns[p];
-
-            // if dofs exist, we have a geometric entity 
-            const std::size_t num = ( (field->getSubcellIndices(d,s).size() > 0) ? 1 : 0 );
+           if ( (patterns[p].first) == FieldType::CG) {
+              RCP<const FieldPattern> field = (patterns[p]).second;
+              // if dofs exist, we have a geometric entity 
+              const std::size_t num = ( (field->getSubcellIndices(d,s).size() > 0) ? 1 : 0 );
+              if(current.size()<num) { 
+                for(int i=num-current.size();i>0;i--,counter++) 
+                  current.push_back(counter);
+              }
+            }
+         } 
+      }
+   }
 
-            if(current.size()<num) { 
-               for(int i=num-current.size();i>0;i--,counter++) 
+   // Add DG fields. These fields are considered internal "cell"
+   // fields for DOF numbering.
+   const int cellDim = dimension_;
+   for(std::size_t d=0;d<dimension_+1;d++) {
+      for(int s=0;s<subcellCount[d];s++) {
+         std::vector<int> & current = patternData_[cellDim][0];
+         for(std::size_t p=0;p<patterns.size();p++) {
+           if ( (patterns[p].first) == FieldType::DG) {
+              RCP<const FieldPattern> field = (patterns[p]).second;
+              // if dofs exist, we have a geometric entity
+              const std::size_t num = ( (field->getSubcellIndices(d,s).size() > 0) ? 1 : 0 );
+              if(current.size()<num) { 
+                for(int i=num-current.size();i>0;i--,counter++) 
                   current.push_back(counter);
+              }
             }
          } 
       }
    }
 
+   // Add SKELETON fields. These fields are considered internal "cell"
```

Comment:
They should live on the interface. The HDG method DOFs on the skeleton are lagrange multipliers enforcing a flux constraint.